jerry-core/api/jerry.c

```diff
@@ -588,15 +632,34 @@ jerry_run (const jerry_value_t func_val) /**< function to run */
     return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG (wrong_args_msg_p)));
   }
 
-  ecma_object_t *func_obj_p = ecma_get_object_from_value (func_val);
+  ecma_object_t *obj_p = ecma_get_object_from_value (func_val);
+
+#if ENABLED (JERRY_MODULE_SYSTEM)
+  if (ecma_object_class_is (obj_p, LIT_MAGIC_STRING_RUNNABLE_UL))
```

Comment:
At the moment there the only runnable is a module object. However if the script parsing is adapted then we should also check the `.extra_info` here to see if it has the ECMA_RUNNABLE_FLAGS_MODULE flag. This check we can already put into this code. Maybe at least as an assert?